src/lint/config/index.ts

```diff
@@ -10,21 +11,17 @@ export type FileLoader = (path: string) => Promise<string | null>;
 export async function discoverConfig(
   fileLoader: FileLoader
 ): Promise<Config | null> {
-  let config = null;
   for (const extension of configFileExtensions) {
     const path = configFileName + extension;
     const configString = await fileLoader(path);
     if (configString === null) continue;
-    try {
-      config = safeLoad(configString);
-    } catch (err) {
-      if (err === YAMLException) continue;
-    }
-    if (typeof config === "object") break;
+    const config = safeLoad(configString);
+    if (typeof config !== "object") continue;
+    const { error } = ConfigSchema.validate(config);
+    if (typeof error !== "undefined") throw new ConfigValidationError(path);
```

Comment:
Can you include the validation errors? So users does not need to review the whole file to know where is the problem.

Reply:
ok, I'll remove the custom error then and use the `ValidationError` that comes with joi